index.js

```diff
@@ -70,10 +70,8 @@ var Block = module.exports = function (data, opts) {
     this.uncleHeaders.push(new BlockHeader(rawUncleHeaders[i], opts))
   }
 
-  // parse transactions
   for (i = 0; i < rawTransactions.length; i++) {
-    var tx = new Tx(rawTransactions[i])
-    tx._homestead = true
+    var tx = new Tx(rawTransactions[i], opts)
```

Comment:
I think this is too dangerous to just pass on the `opts` dictionary since this can have unwanted side effects, e.g. if an options key is named the same on the two libraries (along future changes e.g.). Can you directly pass the instantiated `Common` object, so `{ common: this._common}`?

Reply:
Oops, I missed this. It's a very good point

Reply:
https://github.com/ethereumjs/ethereumjs-block/pull/69/commits/0810af6ada407507160aaec71e897a22e450e922

I did that initially in this commit, but ran into issues with the tests.

Reply:
Ok, we have to figure this out.

Reply:
If you know the solution to 
`Error: Method called with neither a hardfork set nor provided by param
    at Common._chooseHardfork (/Users/micah/dev/ethereumjs-block/node_modules/ethereumjs-common/dist/index.js:114:23)`

Then I can switch to using that.

Reply:
I figured this out. The problem is that if no `Common` object is provided, a new one is created without a `hardfork` param, and this results in an error.

In order to fix this, the else branch of the `if` that sets `this._common` should be something like:

```ts
    let chain = opts.chain ? opts.chain : 'mainnet'

    const tmpCommon = new Common(chain)
    const blockNumber = 123 // TODO: How do we get the block number here?
    
    let hardfork = opts.hardfork ? opts.hardfork : tmpCommon.activeHardfork(blockNumber)
    this._common = new Common(chain, hardfork)
```

Reply:
this reminds me of my initial fix, https://github.com/ethereumjs/ethereumjs-block/pull/68/files

Seems like we'll need the common hardfork to be set after `this.header`